addons/delivery/models/stock_picking.py

```diff
@@ -25,6 +26,8 @@ def _compute_weight(self):
                     weight += ml.product_uom_id._compute_quantity(
                         ml.qty_done, ml.product_id.uom_id) * ml.product_id.weight
             else:
+                if package.package_type_id:
+                    weight = package.package_type_id.base_weight
```

Comment:
you can delete this now that you have the value being set outside of the if/else statement